### ISSUE TITLE
NFC and ISO 7816 support

### ISSUE THREAD

**Issue by 1Dragoon** [go-piv/piv-go#110]
With the current implementation, it doesn't seem practical detect whether we're dealing with a yubikey via NFC, as the only way we can really do so is by looking at the reader name. But that doesn't work if it's over an NFC reader, which will have a different name under PCSC. I think the ideal way to do this would be to look at the card's ATR, which the current PCSC code doesn't support. I've added Windows support for that, (here: https://github.com/1Dragoon/piv-go) though I don't have a means of easily testing any support I add for other platforms.

Once the ATR is obtained, based on this:

http://ludovic.rousseau.free.fr/softwares/pcsc-tools/smartcard_list.txt

It looks like all yubikeys will have some variation of the word "yubikey" in the historical bytes field, sometimes with the first or last y omitted.

**Comment by ericchiang** [comment]
Sorry, I'm not super familiar with ATR

Are you looking for information returned by SCardStatus()? https://pcsclite.apdu.fr/api/group__API.html#gae49c3c894ad7ac12a5b896bde70d0382

Do you have a use case in mind for how you'd use this information? How would exposing this cause you to make different decisions in your application?

**Comment by 1Dragoon** [comment]
Yes that is correct, namely because the ATR is going to tell you what state the card is in, in addition to giving you hints as to what it even is. This is useful in situations where your card may be something other than a yubikey, such as another PIV device or maybe even a seos badge.

Please see the diff here:

https://github.com/go-piv/piv-go/compare/master...1Dragoon:piv-go:master

The main problem I have with support on the unixes is I don't know golang well enough to build definitions for the pcsc-lite library.

**Comment by 1Dragoon** [comment]
Just added the unix side, works on linux, not sure about other platforms as I can't really test any of them. Submitted PR #111
